cli/foundation-models/system/inference/image-classification/prepare_data.py

```diff
@@ -34,43 +44,75 @@ def download_and_unzip(dataset_parent_dir: str, is_multilabel_dataset: int):
         print("done")
     # delete zip file
     os.remove(data_file)
+    return dataset_dir
 
 
 def read_image(image_path):
     with open(image_path, "rb") as f:
         return f.read()
 
-if __name__ == "__main__":
-    parser = argparse.ArgumentParser(
-        description="Prepare data for image classification"
-    )
-    parser.add_argument(
-        "--data_path", type=str, default="./data", help="Dataset location"
-    )
-    parser.add_argument(
-        "--is_multilabel", type=int, default=0, help="Is multilabel dataset"
-    )
-
-    args, unknown = parser.parse_known_args()
-    args_dict = vars(args)
 
-    download_and_unzip(
-        dataset_parent_dir=args.data_path,
-        is_multilabel_dataset=args.is_multilabel,
-    )
-
-    if args.is_multilabel == 0:
-        sample_image = os.path.join(args.data_path, "fridgeObjects", "milk_bottle", "99.jpg")
+def prepare_data_for_online_inference(dataset_dir: str, is_multilabel: int = 0):
+    """Prepare request json for online inference"""
+    if is_multilabel == 0:
+        sample_image = os.path.join(dataset_dir, "milk_bottle", "99.jpg")
     else:
-        sample_image = os.path.join(args.data_path, "multilabelFridgeObjects", "images", "56.jpg")
+        sample_image = os.path.join(dataset_dir, "images", "56.jpg")
 
     request_json = {
-        "inputs": {
-            "image": [base64.encodebytes(read_image(sample_image)).decode("utf-8")],
+        "input_data": {
+            "columns": ["image"],
+            "index": [0],
+            "data": [base64.encodebytes(read_image(sample_image)).decode("utf-8")],
         }
     }
 
-    request_file_name = "sample_request_data.json"
+    request_file_name = os.path.join(dataset_dir, "sample_request_data.json")
 
     with open(request_file_name, "w") as request_file:
-        json.dump(request_json, request_file)
\ No newline at end of file
+        json.dump(request_json, request_file)
+
+
+def prepare_data_for_batch_inference(dataset_dir: str, is_multilabel: int = 0):
+    """Prepare image folder and csv file for batch inference.
+
+    This function will move all images to a single image folder and also create a csv
+    file with images in base64 format.
+    """
+    image_list = []
+
+    csv_file_name = "ml_image_list.csv" if is_multilabel == 1 else "mc_image_list.csv"
```

Comment:
nit: better to call it "image_classification_multilabel_lis.csv" and "image_classification_multiclass_list.csv"

Reply:
Changed.